mempool/txpool/legacypool/legacypool.go

```diff
@@ -141,6 +143,22 @@ type BlockChain interface {
 	StateAt(root common.Hash) (vm.StateDB, error)
 }
 
+// Rechecker defines the minimal set of methods needed to recheck transactions
+// and manage the context that the transactions are rechecked against.
+type Rechecker interface {
+	// GetContext gets the current context that transactions should be rechecked against,
+	// and a way to update that context with changes.
+	GetContext() (sdk.Context, func())
```

Comment:
Comment was a bit confusing imo

```suggestion
	GetContext() (context sdk.Context, write func())
```